UnleashClient/__init__.py

```diff
@@ -46,6 +51,37 @@
 ]
 
 
+def build_ready_callback(
+    event_callback: Optional[Callable[[BaseEvent], None]] = None,
+) -> Optional[Callable]:
+    """
+    Builds a callback function that can be used to notify when the Unleash client is ready.
+    """
+
+    if not event_callback:
+        return None
```

Comment:
Poor mans optional.map()

Reply:
I wonder if it makes sense to support this optional parameter. IMO, if you're going to call this function, you better have a callback. If you supply something that's not a callback then you should expect an error. But having this ability to call all functions with optional and checking if it's set seems a bit off.

On the other hand, if this function can be called from multiple places it centralizes the check in one single place in the code, so just picking your brain about it.